run.py

```diff
@@ -0,0 +1,528 @@
+                "aparcstats2table",
+                "-s",
+                subject_id,
+                f"--hemi={hh}",
+                f"--delimiter=comma",
+                f"--parc={pp}",
+                f"--tablefile={OUTPUT_DIR}/{subject_id}_{hh}_{pp}"
+                + "_stats_area_mm2.csv",
+            ]
+            exec_command(cmd, environ=environ, dry_run=dry_run, cont_output=True)
+
+
+def main(gtk_context):
+
+    fw = gtk_context.client
+
+    log = gtk_context.log
+
+    config = gtk_context.config
+    dry_run = config.get("gear-dry-run")
+
+    anat_dir = INPUT_DIR / "anatomical"
+    anat_dir_2 = INPUT_DIR / "t1w_anatomical_2"
+    anat_dir_3 = INPUT_DIR / "t1w_anatomical_3"
+    anat_dir_4 = INPUT_DIR / "t1w_anatomical_4"
+    anat_dir_5 = INPUT_DIR / "t1w_anatomical_5"
+    t2_dir = INPUT_DIR / "t2w_anatomical"
+
+    # Keep a list of errors and warning to print all in one place at end of log
+    # Any errors will prevent the command from running and will cause exit(1)
+    errors = []
+    warnings = []
+
+    # get # cpu's to set -openmp
+    os_cpu_count = os.cpu_count()
+    log.info("os.cpu_count() = %d", os_cpu_count)
+    n_cpus = config.get("n_cpus")
+    if n_cpus:
+        del config["n_cpus"]
+        if n_cpus > os_cpu_count:
+            log.warning("n_cpus > number available, using %d", os_cpu_count)
+            config["openmp"] = os_cpu_count
+        elif n_cpus == 0:
+            log.info("n_cpus == 0, using %d (maximum available)", os_cpu_count)
+            config["openmp"] = os_cpu_count
+    else:  # Default is to use all cpus available
+        config["openmp"] = os_cpu_count  # zoom zoom
+
+    # grab environment for gear (saved in Dockerfile)
+    with open("/tmp/gear_environ.json", "r") as f:
+        environ = json.load(f)
+
+        # Add environment to log if debugging
+        kv = ""
+        for k, v in environ.items():
+            kv += k + "=" + v + " "
+        log.debug("Environment: " + kv)
+
+    # get config for command by skipping gear config parameters
+    command_config = {}
+    for key, val in config.items():
+        if not key.startswith("gear-"):
+            command_config[key] = val
+    print("command_config:", json.dumps(command_config, indent=4))
+
+    # Validate the command parameter dictionary - make sure everything is
+    # ready to run so errors will appear before launching the actual gear
+    # code.  Add descriptions of problems to errors & warnings lists.
+    # print("gtk_context.config:", json.dumps(gtk_context.config, indent=4))
+
+    # The main command line command to be run:
+    command = ["time", "recon-all"]
+
+    if Path(LICENSE_FILE).exists():
+        log.debug("%s exists.", LICENSE_FILE)
+    install_freesurfer_license(gtk_context, LICENSE_FILE)
+
+    subject_id = config.get("subject_id")
+    if not subject_id:
+        subject_id = fw.get_analysis(gtk_context.destination["id"]).parents.subject
+        subject = fw.get(subject_id)
```

Comment:
Yes, good catch.  Roy said it does save time because the generic `get` searches for the id in all sorts of places instead of starting at the right place.  I'll look through and change all of these.